Tegu-class Communications Freighter

goober3 opened the pull request:
## About The Pull Request
This PR adds a brand new ship to Nanotrasen's post-rework roster.

<details><summary>Images</summary>

![68747470733a2f2f616666656374656461726330372e626c6f622e636f72652e77696e646f77732e6e65742f6d6462322f696d616765732f3336373235323037352f33333231373933373937312f612f5f6d6170735f73687574746c65735f6e616e6f74726173656e5f](https://github.com/user-attachments/assets/a2c95f8d-fab7-4f16-ba47-ce1b549fd6b3)

**images below are outdated**
![image](https://github.com/shiptest-ss13/Shiptest/assets/118859017/6a41eddd-fcbe-4630-b1b9-814d3f966671)

![image](https://github.com/shiptest-ss13/Shiptest/assets/118859017/39594bbe-7837-4d49-a21f-442e42af5e2d)

</details>

The Tegu-class is an engineering freighter equipped with a full telecommunications setup and a subshuttle. It is built to not only facilitate communications between Nanotrasen-aligned vessels in the sector, but also foster several smaller networks for other vessels with a small array of relays. The ship also comes equipped with a small cargo department, an engineering department, a small communications room, and a supply office for the QM.

This PR also adds a new design disk for printing telecommunications parts. 

To-do:
- [x] Ship description.
- [ ] In-game testing.
- [x] Add security role.
- [ ] Mining facilities.

## Why It's Good For The Game
more ship

## Changelog

🆑goober3, Cloudbreak
add: The year is 2070. The Tegu-class Communications Freighter has just been fullmerged.
add: A new subshuttle, the Skink-class Cargo Runner.
/🆑

goober3 committed:
commit number one

goober3 committed:
linter fix !!

goober3 committed:
unconnected vents & scrubbers

goober3 committed:
Merge branch 'master' into ship-teguu

EEASAS commented:
okkk.... but isnt there like.... new ship restrictions.... like you cant make new nanotrasen/independent/syndicate/srm/solgov ships......

meemofcourse commented:
do they know chat

goober3 commented:
> okkk.... but isnt there like.... new ship restrictions.... like you cant make new nanotrasen/independent/syndicate/srm/solgov ships......

i'm doing this in defiance of the maptainers because i hate them

thgvr commented:
> okkk.... but isnt there like.... new ship restrictions.... like you cant make new nanotrasen/independent/syndicate/srm/solgov ships......

This restriction is for contributors outside of the main team.

rye-rice commented:
sowwy, nwo prwogwess iwn mwonths, rweopwen wwen wrorking on iwt agwian

goober3 committed:
Merge remote-tracking branch 'upstream/master' into ship-teguu

goober3 commented:
we're so back

goober3 committed:
Merge branch 'master' into ship-teguu

goober3 committed:
Merge branch 'ship-teguu' of https://github.com/goober3/hi-github-portside into ship-teguu

goober3 committed:
comms tweaks, thank you cloudbreak

goober3 committed:
Merge branch 'master' into ship-teguu

thgvr commented:
I am begging you to finish this

goober3 commented:
> I am begging you to finish this

this is technically already finished. it just needs a review and a description.

a telecomms guide too maybe

meemofcourse reviewed (changes requested):
![imagen](https://github.com/user-attachments/assets/ac9c11fa-a89b-477d-9cc9-d6c52f89567f)
i do not like that the lightswitch in the officer's lounge controls the lights in the officers' quarters. there are also three light switches in the cargo office
![imagen](https://github.com/user-attachments/assets/82bf8cdc-0b34-4c78-a904-c6be7c3f9083)
this restroom needs better lighting
![imagen](https://github.com/user-attachments/assets/bd704628-5fe9-4a45-a4b6-94f190ee3cba)
i believe the pda painter is a centcom utility and it lets you have SUNS pdas so it would be best to remove it
![imagen](https://github.com/user-attachments/assets/7db5319e-ee04-405f-9de9-4ac53599a318)
the only set of tools i could find in this ship are hidden below two welding helmets. consider adding toolboxes and toolbelts to the engineer and engi director lockers - they should not start with them
![imagen](https://github.com/user-attachments/assets/b260f892-62dc-4247-9ec6-7e67d95c87b3)
maybe have an emergency suit here
![imagen](https://github.com/user-attachments/assets/2dd53e5b-72bc-4b9f-ae05-2eec90b9cb2f)
while i'm not an engi player i don't think this is right

maybe a ship like this could use a power monitor. i'd ask a maptainer/other person who is more experienced with telecommunications to test things out but cool ship

goober3 commented:
> i do not like that the lightswitch in the officer's lounge controls the lights in the officers' quarters. there are also three light switches in the cargo office

this was done so i wouldn't have to make more areas and add APCs to the officer's quarters but in hindsight i probably should just do that. also the multiple lightswitches exist for the same reason you'd have two light switches on both ends of a hallway. it's more convenient like that.

> i believe the pda painter is a centcom utility and it lets you have SUNS pdas so it would be best to remove it

i might add a generic sanitised version of the PDA painter later in a seperate PR. for now they can live without.

thak yuo

goober3 committed:
see commit description

-added ship description thank you apogee
-made officer quarters unique areas
-replaced non-bridge computers with retro computers
-added power monitor
-added emergency suits
-added an extra light to the bathroom
-added toolbelts + boxes to all engineer lockers, except for an extra CE belt because the CE spawns with a set already and i don't want multiple here
-replaced the props on the supply director's desk with a book and pencil because the lamp clipped with the window
-added the electrical closet back because PDA painters allow painting of faction PDAs

goober3 committed:
Merge branch 'master' into ship-teguu

goober3 committed:
unconnected scrubber jumpscare

goober3 committed:
Merge branch 'ship-teguu' of https://github.com/goober3/hi-github-portside into ship-teguu

goober3 committed:
i forgor to make the captain's quarters a unique area

fighterslam commented:
The "/turf/open/floor/plasteel/telecomms_floor" (north floor) starts with the normal, default airmix, 101.3 KPA, etc. The circuit floors (south floor) "turf/open/floor/circuit/telecomms" constantly produce a colder airmix, causing these two adjacent turfs to have atmos differences and process once they spawn in, so I think this might be a small performance concern. It was something I had to deal with on the Ranger too - you probably either want to set the circuit floors to normal circuit floors so there's no atmos differences, or make a new telecomms_floor type that has the same initial airmix. 
![image](https://github.com/user-attachments/assets/a56a1362-29a7-4ba3-8c34-394d34aab938)

meemofcourse reviewed (approved):
this ship rocks dude

rye-rice commented:
@shiptest-ss13/maptainers

Sadhorizon commented on _maps/configs/nanotrasen_tegu.json:
```diff
@@ -0,0 +1,47 @@
+{
```
Add `"faction": "/datum/faction/nt"`, somewhere, or it's gonna show as indie on the spawning menu.

goober3 committed:
Merge remote-tracking branch 'upstream/master' into ship-teguu

goober3 committed:
im stuff

goober3 committed:
no need for this now

thgvr commented:
@shiptest-ss13/maptainers ancient pr, please decide if its being merged or not

goober3 commented:
> @shiptest-ss13/maptainers ancient pr, please decide if its being merged or not

this still has yet to be tested substantially in-game actually. some day...

DorfGuy1 commented:
Where's the guide?
 TO the comms

goober3 committed:
Merge remote-tracking branch 'upstream/master' into ship-teguu

goober3 committed:
see desc

-fixed turrets
-fixed paths
-altered the hull to make it more symmetrical
-fixed shuttle access
-increased ship funds by 2000 credits

goober3 committed:
replaces light turrets with medium turrets

goober3 committed:
i'm a doofus

goober3 committed:
VI security slot

goober3 committed:
apc fix

goober3 committed:
moves sec vent & replaces canteen coffee vendor with soda vendor

tzui5020 commented:
It needs a chair or a few railings in the Cargobay please. I beg you

goober3 committed:
mining conversion

thgvr commented:
purgatory

meemofcourse reviewed (approved):
BWAAH SCARY TELECOMMS
